### ISSUE TITLE
[BUG] Card Component response

### ISSUE THREAD

**Issue by nsimunko** [Adyen/adyen-ios#249]
I am integrating the `CardComponent` into an iOS app. In the `clientKey` property I send the encryption key that I get from my backend. That same key Android uses as `publicKey`, and they get the encrypted card data back. 

My app fails in the `CardComponentExtensions` file, line 34, `fetchCardPublicKey` method. 
If I'm not mistaken, my `clientKey` should like this: `test_870be...`, but it looks like this: `10001|C34C...`. 

If it works on Android, will it also be removed in a future release? 
Do I need to get my key from the Customer Area? 

The response I get from the `fetchCardPublicKey` methods fails to decode, so I don't even get a "Public key bad" error. It just fails to decode because it's not a valid JSON response. 

The response I get looks like this: `/checkoutshopper/v1/clientKeys/10001|C34C...`, so response parsing fails in `APIClient.swift`, line 81, on `let response = try Coder.decode(data) as R.ResponseType`.

The error is: 
```
DecodingError
  ▿ dataCorrupted : Context
    - codingPath : 0 elements
    - debugDescription : "The given data was not valid JSON."
    ▿ underlyingError : Optional<Error>
      - some : Error Domain=NSCocoaErrorDomain Code=3840 "No value." UserInfo={NSDebugDescription=No value.}
```

I get that I probably need a new key, to pass as a `clientKey`, but I don't think that despite an invalid `clientKey` decoding should fail like it does in my case 😄  
At least give me some error with a message, pls 🙏  

I just the encrypted card data back through the `PaymentComponentDelegate`. 

Reproducing is pretty straight forward. Put the api encryption key in the `CardComponent` init.

Cheers.

**Comment by mohammedDehairy** [comment]
Hi @nsimunko ,

 Thanks for reaching out.

 The client key for test environment indeed should be `test_870be...`, `10001|C34C...` is the card public key not the client key.

 Please follow [the docs](https://docs.adyen.com/development-resources/client-side-authentication#get-your-client-key) to get the client key corresponding to your account.

**Comment by nsimunko** [comment]
Thanks for the response @mohammedDehairy, 
I will ask for my backend team to give me the client key.
What about the decoding error i mentioned? Should I receive an error response in a JSON format? 

I opened this issue mainly because it looks like that shouldn't happen. I can catch this specific error message and throw a "invalid encryption key" alert.

**Comment by mohammedDehairy** [comment]
@nsimunko ,

You're right, a proper json error message makes more sense, I'll reach out to our backend team and see what we can do.

**Comment by mohammedDehairy** [comment]
@nsimunko ,

 can you provide the full html response sent? it should have been logged in the console, if `AdyenLogging.isEnabled` is true.

**Comment by nsimunko** [comment]
When I submit a card with the public key entered as client key, I get this in the console. That's it. No HTML. I get some HTML when I open an empty form without an empty string as client key. But this is all I see when I submit a valid Test card, with an invalid key. 
```
---- Request (/checkoutshopper/v1/clientKeys/10001|C34C06...) ----
{
```

```
---- Response (/checkoutshopper/v1/clientKeys/10001|C34C06...) ----
```

This is how I initialize an empty card component, which should just give me the an encrypted card object, which I will send to my backend. 
```
let cardTypes: [CardType] = [.americanExpress, .diners, .maestro, .masterCard, .visa]

// TestPaymentMethod is just a AnyCardPaymentMethod
let paymentMethod = TestPaymentMethod(brands: cardTypes.map{ $0.rawValue }, fundingSource: .credit, type: "credit", name: "TestCard")

let style = AdyenStyleManager.formStyle(for: theme)
        
let card = CardComponent(paymentMethod: paymentMethod, clientKey: clientKey, style: style)
        
card.supportedCardTypes = [.americanExpress, .diners, .maestro, .masterCard, .visa]
card.clientKey = clientKey
card.showsLargeTitle = false
card.showsStorePaymentMethodField = false
card.showsHolderNameField = true
card.environment = .test
        
return card
```

**Comment by mohammedDehairy** [comment]
Thanks @nsimunko ,

 I have created an internal ticket, and it will be fixed in a future release of the Adyen Checkout backend API.

**Comment by nsimunko** [comment]
Just wanted to let you know that my card component works now, with the new client key. 
Hope that my issue has helped you somehow. 👍

**Comment by descorp** [comment]
We have improved our error handling in HTTP client. Released in 3.8.1

Thanks @nsimunko ! 
This is definitely going to help many!

I am going to close this ticket for now, feel free to provide any additional comments in this thread.
